interpreter/k8s/src/main/scala/io/buoyant/interpreter/k8s/ConfigMapInterpreterInitializer.scala

```diff
@@ -0,0 +1,101 @@
+package io.buoyant.interpreter
+package k8s
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.Dtab
+import com.twitter.finagle.Stack.Params
+import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.finagle.param.Label
+import com.twitter.logging.Logger
+import com.twitter.util._
+import io.buoyant.config.types.Port
+import io.buoyant.k8s.ClientConfig
+import io.buoyant.k8s.v1._
+import io.buoyant.namer.{ConfiguredDtabNamer, InterpreterConfig, InterpreterInitializer, Param}
+import java.util.concurrent.atomic.AtomicReference
+
+class ConfigMapInterpreterInitializer extends InterpreterInitializer {
+  val configClass = classOf[ConfigMapInterpreterConfig]
+  override def configId: String = "io.l5d.k8s.configMap"
+}
+
+object ConfigMapInterpreterInitializer extends ConfigMapInterpreterInitializer
+
+case class ConfigMapInterpreterConfig(
+  host: Option[String],
+  port: Option[Port],
+  namespace: Option[String],
+  name: String,
+  filename: String
+) extends InterpreterConfig with ClientConfig {
+
+  @JsonIgnore
+  override def experimentalRequired = true
+
+  @JsonIgnore
+  def portNum = port.map(_.port)
+
+  @JsonIgnore
+  private[this] val log = Logger()
+
+  @JsonIgnore
+  val nsOrDefault = namespace.getOrElse(DefaultNamespace)
+
+  @JsonIgnore
+  val api = {
```

Comment:
Totally, yeah, this stuff is pretty icky. I would take a look at the implementations of [IngressCache](https://github.com/linkerd/linkerd/blob/master/k8s/src/main/scala/io/buoyant/k8s/IngressCache.scala#L81) and [K8sDtabStore], since that's where I cribbed from when putting this branch together in the first place. And don't stress too much if there isn't an easy way to untangle this. I just wanted to give it some attention because it's not super user friendly at the moment.